src/main/java/org/mskcc/oncokb/curation/importer/CanonicalTranscriptFlagImporter.java

```diff
@@ -0,0 +1,120 @@
+package org.mskcc.oncokb.curation.importer;
+
+import static org.mskcc.oncokb.curation.util.FileUtils.readDelimitedLinesStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.mskcc.oncokb.curation.domain.Flag;
+import org.mskcc.oncokb.curation.domain.Transcript;
+import org.mskcc.oncokb.curation.domain.enumeration.FlagType;
+import org.mskcc.oncokb.curation.domain.enumeration.ReferenceGenome;
+import org.mskcc.oncokb.curation.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * To backfill the OncoKB canonical transcripts
+ */
+
+@Component
+public class CanonicalTranscriptFlagImporter {
+
+    private final Logger log = LoggerFactory.getLogger(CanonicalTranscriptFlagImporter.class);
+
+    final String ONCOKB_GENE_LIST_FILE_PATH = "data/oncokb_core/cancer_gene_list_v4_23.tsv";
+
+    final GeneService geneService;
+    final TranscriptService transcriptService;
+    final FlagService flagService;
+
+    public CanonicalTranscriptFlagImporter(
+        GeneService geneService,
+        GenomeNexusService genomeNexusService,
+        EnsemblGeneService ensemblGeneService,
+        TranscriptService transcriptService,
+        FlagService flagService,
+        InfoService infoService,
+        MainService mainService
+    ) {
+        this.geneService = geneService;
+        this.transcriptService = transcriptService;
+        this.flagService = flagService;
+    }
+
+    public void importCanonicalTranscripts() {
+        URL geneListUrl = getClass().getClassLoader().getResource(ONCOKB_GENE_LIST_FILE_PATH);
+        try {
+            InputStream is = geneListUrl.openStream();
+            List<List<String>> tsvFile = readDelimitedLinesStream(is, "\t", true);
+            for (int i = 0; i < tsvFile.size(); i++) {
+                if (i == 0) {
+                    continue;
+                }
+                List<String> line = tsvFile.get(i);
+                if (line.size() < 8) {
+                    log.warn(String.format("Skipping line %i", i));
+                    continue;
```

Comment:
Is the data corrupted if this happens?

Reply:
Shouldn't happen unless the cancer gene list file we generated in oncokb-data repo is corrupted.

Reply:
In that case I think it should throw an exception. The file is in source control so there shouldn't be a reason why it would not fail locally, but only in production. If it's a warning it's easy to miss.